tests/integ/attr_test.py

```diff
@@ -1454,6 +1588,526 @@ def testNaNAttributeValue(self):
+        self.assertEqual(rsp.status_code, 201)  # this is ok
+
+        # get all attributes and verify the one we created is there
+        expected_type = {'class': 'H5T_FLOAT', 'base': 'H5T_IEEE_F32LE'}
+        expected_shape = {'class': 'H5S_SIMPLE', 'dims': [6]}
+        params = {"IncludeData": 1}
+        rsp = self.session.get(req, headers=headers, params=params)
+        self.assertEqual(rsp.status_code, 200)
+        rspJson = json.loads(rsp.text)
+        self.assertTrue("hrefs" in rspJson)
+        self.assertTrue("attributes" in rspJson)
+        rsp_attributes = rspJson["attributes"]
+        self.assertEqual(len(rsp_attributes), 1)
+        rsp_attr = rsp_attributes[0]
+        self.assertTrue("name" in rsp_attr)
+        self.assertEqual(rsp_attr["name"], attr_name)
+        self.assertTrue("href" in rsp_attr)
+        self.assertTrue("created" in rsp_attr)
+        self.assertTrue("type" in rsp_attr)
+        self.assertEqual(rsp_attr["type"], expected_type)
+        self.assertTrue("shape" in rsp_attr)
+        self.assertTrue(rsp_attr["shape"], expected_shape)
+        self.assertTrue("value" in rsp_attr)
+        self.assertEqual(rsp_attr["value"], value)
+
+        # try doing a get on this specific attribute
+        rsp = self.session.get(bad_req, headers=headers)
+        self.assertEqual(rsp.status_code, 404)  # can't do a get with the attribute name
+
+        # do a post request with the attribute name
+        attr_names = [attr_name, ]
+        data = {"attr_names": attr_names}
+        rsp = self.session.post(req, data=json.dumps(data), params=params, headers=headers)
+        self.assertEqual(rsp.status_code, 200)
+        rspJson = json.loads(rsp.text)
+        self.assertTrue("attributes" in rspJson)
+        rsp_attributes = rspJson["attributes"]
+        self.assertEqual(len(rsp_attributes), 1)
+        rsp_attr = rsp_attributes[0]
+
+        self.assertTrue("name" in rsp_attr)
+        self.assertEqual(rsp_attr["name"], attr_name)
+        self.assertTrue("href" in rsp_attr)
+        self.assertTrue("created" in rsp_attr)
+        self.assertTrue("type" in rsp_attr)
+        self.assertEqual(rsp_attr["type"], expected_type)
+        self.assertTrue("shape" in rsp_attr)
+        self.assertTrue(rsp_attr["shape"], expected_shape)
+        self.assertTrue("value" in rsp_attr)
+        self.assertEqual(rsp_attr["value"], value)
+
+        # try deleting the attribute by name
+        rsp = self.session.delete(bad_req, headers=headers)
+        self.assertEqual(rsp.status_code, 404)  # not found
+
+        # send attribute name as an encoded query param
+        attr_names_param = base64.b64encode(attr_name.encode("utf8")).decode("ascii")
+        # specify a seperator since our attribute name has the default slash
+        params = {"attr_names": attr_names_param, "encoding": "base64", "seperator": "!"}
+        rsp = self.session.delete(req, params=params, headers=headers)
```

Comment:
I think we should include a test that deletes multiple attributes at once through `DELETE_Attributes`.  I wrote the following while testing the API:

```
    def testDeleteAttributesMultiple(self):
        print("testDeleteAttributesMultiple", self.base_domain)
        headers = helper.getRequestHeaders(domain=self.base_domain)
        req = self.endpoint + "/"

        # Get root uuid
        rsp = self.session.get(req, headers=headers)
        self.assertEqual(rsp.status_code, 200)
        rspJson = json.loads(rsp.text)
        root_uuid = rspJson["root"]
        helper.validateId(root_uuid)

        # create a subgroup
        req = self.endpoint + "/groups"
        rsp = self.session.post(req, headers=headers)
        self.assertEqual(rsp.status_code, 201)
        rspJson = json.loads(rsp.text)
        grp_id = rspJson["id"]
        self.assertTrue(helper.validateId(grp_id))

        # link as "grp1"
        grp_name = "grp1"
        req = self.endpoint + "/groups/" + root_uuid + "/links/" + grp_name
        payload = {"id": grp_id}
        rsp = self.session.put(req, data=json.dumps(payload), headers=headers)
        self.assertEqual(rsp.status_code, 201)  # created

        # Create attributes
        for i in range(10):
            attr_name = "attr" + str(i * 100)
            value = [i]
            data = {"type": "H5T_IEEE_F32LE", "shape": 1, "value": value}
            req = self.endpoint + "/groups/" + grp_id + "/attributes/" + attr_name
            rsp = self.session.put(req, data=json.dumps(data), headers=headers)
            self.assertEqual(rsp.status_code, 201) # Created

        # Delete all by parameter
        attr_names = ["attr" + str(i * 100) for i in range(10)]
        separator = '/'

        req = self.endpoint + "/groups/" + grp_id + "/attributes?attr_names=" + separator.join(attr_names)
        rsp = self.session.delete(req, headers=headers)
        self.assertEqual(rsp.status_code, 200)

        # Attempt to read deleted attributes
        for i in range(10):
            req = self.endpoint + "/groups/" + grp_id + "/attributes/" + "attr" + str(i)
            rsp = self.session.get(req, headers=headers)
            self.assertEqual(rsp.status_code, 404)

        # Create another batch of attributes
        for i in range(10):
            attr_name = "attr" + str(i * 100)
            value = [i]
            data = {"type": "H5T_IEEE_F32LE", "shape": 1, "value": value}
            req = self.endpoint + "/groups/" + grp_id + "/attributes/" + attr_name
            rsp = self.session.put(req, data=json.dumps(data), headers=headers)
            self.assertEqual(rsp.status_code, 201) # Created

        # Delete with custom separator
        attr_names = ["attr" + str(i * 100) for i in range(10)]
        separator = ':'

        req = self.endpoint + "/groups/" + grp_id + "/attributes?attr_names=" + separator.join(attr_names) + "&seperator=" + separator
        rsp = self.session.delete(req, headers=headers)
        self.assertEqual(rsp.status_code, 200)

        # Attempt to read
        for i in range(10):
            req = self.endpoint + "/groups/" + grp_id + "/attributes/" + "attr" + str(i)
            rsp = self.session.get(req, headers=headers)
            self.assertEqual(rsp.status_code, 404)
```

Reply:
Good idea. Added test